addons/purchase/models/purchase.py

```diff
@@ -536,10 +536,10 @@ def _compute_qty_received(self):
             total = 0.0
             for move in line.move_ids:
                 if move.state == 'done':
-                    if move.product_uom != line.product_uom:
+                    if move.location_dest_id.usage == "internal":
                         total += move.product_uom._compute_quantity(move.product_uom_qty, line.product_uom)
-                    else:
-                        total += move.product_uom_qty
+                    elif move.location_dest_id.usage == "supplier" and move.to_refund:
+                        total -= move.product_uom._compute_quantity(move.product_uom_qty, line.product_uom)
```

Comment:
no.. as I already explained that condition will increase qty also when we return product without refund and that is wrong.. so i put internal location condition..